lib/cli/src/commands/create_exe.rs

```diff
@@ -0,0 +1,273 @@
+        let wasm_object_path = PathBuf::from("wasm.o");
+        #[cfg(windows)]
+        let wasm_object_path = PathBuf::from("wasm.obj");
+
+        let wasm_module_path = starting_cd.join(&self.path);
+
+        let header_file_path = Path::new("my_wasm.h");
+        let module =
+            Module::from_file(&store, &wasm_module_path).context("failed to compile Wasm")?;
+        let _ = module.serialize_to_file(&wasm_object_path)?;
+
+        let artifact: &wasmer_engine_object_file::ObjectFileArtifact =
+            module.artifact().as_ref().downcast_ref().context(
+                "Engine type is ObjectFile but could not downcast artifact into ObjectFileArtifact",
+            )?;
+        let symbol_registry = artifact.symbol_registry();
+        let metadata_length = artifact.metadata_length();
+        let module_info = module.info();
+        let header_file_src = crate::c_gen::object_file_header::generate_header_file(
+            module_info,
+            symbol_registry,
+            metadata_length,
+        );
+
+        let header_path = header_file_path.clone();
+        // for C code
+        let mut header = std::fs::OpenOptions::new()
+            .create(true)
+            .truncate(true)
+            .write(true)
+            .open(&header_path)?;
+
+        use std::io::Write;
+        header.write(header_file_src.as_bytes())?;
+
+        // auto compilation
+        //
+
+        // write C src to disk
+        let c_src_path = Path::new("wasmer_main.c");
+        #[cfg(not(windows))]
+        let c_src_obj = PathBuf::from("wasmer_main.o");
+        #[cfg(windows)]
+        let c_src_obj = PathBuf::from("wasmer_main.obj");
+
+        {
+            let mut c_src_file = fs::OpenOptions::new()
+                .create_new(true)
+                .write(true)
+                .open(&c_src_path)
+                .context("Failed to open C source code file")?;
+            // TODO:
+            c_src_file.write_all(WASMER_MAIN_C_SOURCE)?;
+        }
+        run_c_compile(&c_src_path, &c_src_obj).context("Failed to compile C source code")?;
+        LinkCode {
+            object_paths: vec![c_src_obj, wasm_object_path],
+            output_path,
+            additional_libraries: self.libraries.clone(),
+            ..Default::default()
+        }
+        .run()
+        .context("Failed to link objects together")?;
+
+        eprintln!(
+            "✔ Native executable compiled successfully to `{}`.",
+            self.output.display(),
+        );
+
+        Ok(())
+    }
+}
+
+fn get_wasmer_include_directory() -> anyhow::Result<PathBuf> {
+    let mut path = PathBuf::from(env::var("WASMER_DIR")?);
+    path.push("include");
+    Ok(path)
+}
+
+fn get_libwasmer_path() -> anyhow::Result<PathBuf> {
+    let mut path = PathBuf::from(env::var("WASMER_DIR")?);
+    path.push("lib");
+
+    #[cfg(not(windows))]
+    path.push("libwasmer.a");
+    #[cfg(windows)]
+    path.push("libwasmer.lib");
+
+    Ok(path)
+}
```

Comment:
It would be a good idea to abstract this out, and also reuse them on the `lib/cli/src/commands/config.rs` file

Reply:
Yeah it looks like the config logic isn't tested right now... changes Ivan and I have been making to the C API are going to break it I think...

The config code implies we can just do `-lwasmer` which is interesting that would make the code have fewer `cfg`s in it.

Reply:
The code in `wasmer config` for linking against wasmer seems to default to the dynamic libwasmer 🤔